docs/source/graph-construction/gs-processing/gpartition/sagemaker.rst

```diff
@@ -0,0 +1,133 @@
+
+Step 2: Upload Docker Images to Amazon ECR Repository
+.......................................................
+Because SageMaker relies on Amazon ECR to access customers' own Docker images, users need to upload Docker images built in the Step 1 to their own ECR repository.
+
+The following command will authenticate the user account to access to user's ECR repository via AWS CLI.
+
+.. code-block:: bash
+
+    aws ecr get-login-password --region <REGION> | docker login --username AWS --password-stdin <AWS_ACCOUNT_ID>.dkr.ecr.<REGION>.amazonaws.com
+
+Please replace the `<REGION>` and `<AWS_ACCOUNT_ID>` with your own account information and be consistent with the values used in the **Step 1**.
+
+In addition, users need to create an ECR repository at the specified `<REGION>` with the name as `<IMAGE_NAME>` **WITHOUT** the ECR URI string, e.g., ``graphstorm``.
+
+And then use the below command to push the built GraphStorm Docker image to users' own ECR repository.
+
+.. code-block:: bash
+
+    docker push <IMAGE_NAME>:<IMAGE_TAG>
+
+Please replace the `<IMAGE_NAME>` and `<IMAGE_TAG>` with the actual Docker image name and tag, e.g., ``888888888888.dkr.ecr.us-east-1.amazonaws.com/graphstorm:sm-gpu``.
+
+Launch the partition job on Amazon SageMaker
+---------------------------------------------
+
+For this example we'll use a same size SageMaker cluster with 2 ``ml.t3.xlarge`` instances
+as in :ref:`GSProcessing Example<gsp-examining-output>`. In this tutorial, we assume the data is already ready on AWS S3 bucket.
+For large graph, users can select larger instance type and more instances in number.
+
+Install dependencies
+.....................
+To run GraphStorm with the Amazon SageMaker service, users should set up an instance with the SageMaker library installed and GraphStorm's SageMaker tools copied.
+
+1. Use the below command to install SageMaker.
+
+.. code-block:: bash
+
+    pip install sagemaker
+
+2. Copy GraphStorm SageMaker tools. Users can clone the GraphStorm repository using the following command or copy the `sagemaker folder <https://github.com/awslabs/graphstorm/tree/main/sagemaker>`_ to the instance.
+
+.. code-block:: bash
+
+    git clone https://github.com/awslabs/graphstorm.git
+
+Launch graph partitioning task
+...............................
+Users can use the following command to launch partition jobs.
+
+.. code:: bash
+
+   python launch/launch_partition.py \
+       --graph-data-s3 ${DATASET_S3_PATH} \
+       --num-parts ${NUM_PARTITIONS} \
+       --instance-count ${NUM_INSTANCES} \
+       --output-data-s3 ${OUTPUT_PATH} \
+       --instance-type ${INSTANCE_TYPE} \
+       --image-url ${IMAGE_URI} \
+       --region ${REGION} \
+       --role ${ROLE}  \
+       --entry-point "run/partition_entry.py" \
+       --metadata-filename ${METADATA_FILE} \
+       --log-level INFO \
+       --partition-algorithm ${ALGORITHM}
+
+.. warning::
+    The ``NUM_INSTANCES`` should be a multiple of ``NUM_PARTITIONS`` here.
```

Comment:
Exact is better, using more instances than num parts I think has no benefit?

Reply:
Why not? I suppose more instances will provide better scalabilities on memory and number of cores.

Reply:
I don't think the implementation supports this scenario.

Reply:
I make the description to exact, I run some simple experiments and did not see any boost here. Let us make it as exact here.